src/jit/ee_il_dll.cpp

```diff
@@ -402,7 +402,12 @@ unsigned CILJit::getMaxIntrinsicSIMDVectorLength(CORJIT_FLAGS cpuCompileFlags)
     if (!jitFlags.IsSet(JitFlags::JIT_FLAG_PREJIT) && jitFlags.IsSet(JitFlags::JIT_FLAG_FEATURE_SIMD) &&
         jitFlags.IsSet(JitFlags::JIT_FLAG_USE_AVX2))
     {
-        if (JitConfig.EnableAVX() != 0 && JitConfig.EnableAVX2() != 0)
+        // Since the ISAs can be disabled individually and since they are hierarchical in nature (that is
+        // disabling SSE also disables SSE2 through AVX2), we need to check each ISA in the hierarchy to
+        // ensure that AVX2 is actually supported. Otherwise, we will end up getting asserts downstream.
+        if ((JitConfig.EnableAVX2() != 0) && (JitConfig.EnableAVX() != 0) && (JitConfig.EnableSSE42() != 0) &&
```

Comment:
This is the drawback to them being hierarchical, we now need to check each configuration knob to determine if AVX2 is actually supported here. This didn't look to be a particularly hot method, so I don't think this will be a concern.

If we have an initialized compiler, we could shortcut this by just checking `JitTls::GetCompiler()->compSupports(InstructionSet_AVX2)` (we would also need to make that method visible outside `Compiler` itself). However, based on the check on L412, it looks like we might not always have a compiler when we are in here.

Reply:
This was causing asserts when disabling `Base` through `SSE42` without also disabling `AVX` or `AVX2`